packages/eslint-plugin/lib/rules/no-aria-hidden-on-focusable.js

```diff
@@ -0,0 +1,107 @@
+/**
+ * @typedef { import("../types").RuleModule<[]> } RuleModule
+ * @typedef { import("@html-eslint/types").Tag } Tag
+ */
+
+const { RULE_CATEGORY } = require("../constants");
+const { findAttr } = require("./utils/node");
+const { createVisitors } = require("./utils/visitors");
+const { getRuleUrl } = require("./utils/rule");
+
+const MESSAGE_IDS = {
+  UNEXPECTED: "unexpected",
+};
+
+// List of elements that are inherently focusable
+const FOCUSABLE_ELEMENTS = new Set([
+  "a", // if href is present
+  "button",
+  "input",
+  "select",
+  "textarea",
+  "video", // if controls is present
+  "audio", // if controls is present
+  "details",
+  "embed",
+  "iframe",
+]);
+
+/**
+ * @type {RuleModule}
+ */
+module.exports = {
+  meta: {
+    type: "code",
+
+    docs: {
+      description: 'Disallow aria-hidden="true" on focusable elements',
+      category: RULE_CATEGORY.ACCESSIBILITY,
+      recommended: false,
+      url: getRuleUrl("no-aria-hidden-on-focusable"),
+    },
+
+    fixable: null,
+    schema: [],
+    messages: {
+      [MESSAGE_IDS.UNEXPECTED]:
+        'Unexpected aria-hidden="true" on focusable element.',
+    },
+  },
+
+  create(context) {
+    /**
+     * Checks if an element is focusable
+     * @param {Tag} node
+     * @returns {boolean}
+     */
+    function isFocusable(node) {
+      const tagName = node.name.toLowerCase();
+
+      // Check for tabindex attribute
+      const tabIndexAttr = findAttr(node, "tabindex");
+      if (tabIndexAttr && tabIndexAttr.value) {
+        const tabIndexValue = tabIndexAttr.value.value;
+        // If tabindex is -1, the element is not focusable
+        if (tabIndexValue === "-1") {
+          return false;
+        }
+        // If tabindex is present and not -1, the element is focusable
+        return true;
+      }
```

Comment:
If the `contenteditable` attribute is an `empty string (“”)`, `"true"` or `"plaintext-only"`, the element will also become focusable. You can do something similar with `tabindex`.